File: Richie/Linked List Cycle/linked-list-cycle.py
class Solution:
    def hasCycle(self, head):
        try:
            fast = head.next.next
            slow = head.next
            while head.next:       
                if fast == slow:
                    return True

                slow = slow.next
                fast = fast.next.next
            return False
        except:
            return False

File: Engr-T.stark/linkedListCycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        unordered_map<ListNode*, bool> visit;
        while(head){
            if(visit[head]) {
               return true; 
            }
            visit[head] = true;
            head = head->next;
        }
        return false;
    }
};
